### PR TITLE
feat: upload assets to the release associated with given tagName

### PR DESCRIPTION
**Pull request by chiichen**
Enable uploading assets to the target release associated with the given tagName (if exist). 

## TL;DR
The basic motivation is that I'm using [googleapis/release-please](https://github.com/googleapis/release-please) github action to release automatically. However, it only provides `tagName` as output if a release is created successfully. And we can use [this](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-a-release-by-tag-name) rest api to get the corresponding releaseId by the given tagName. That's why I did it. 

## Test Example

I've apply this action on my project here [Chiichen/JinhsiStudio](https://github.com/Chiichen/JinhsiStudio). This is the related [action](https://github.com/Chiichen/JinhsiStudio/actions/runs/10528090151) and the [release](https://github.com/Chiichen/JinhsiStudio/releases/tag/jinhsi-studio-v0.2.0). It seems to be working well

Let me know if it's an acceptable enhancement. And I will add some docs for it (README or somewhere else?)

### CODE REVIEW

**Comment by chiichen** [comment]
I use minor version since It maintains forward compatibility.

**Comment by chiichen** [comment]
@FabianLars Friendly ping. Any comments for it?

**Comment by FabianLars** [comment]
yeah sorry, i wanted to look at this for ages but there's always some bs coming in between 🙃 

I hope i can give it a proper look on monday or tuesday (will be on vacation after that), but the thing i already noticed is that there's quite a lot of duplicate code now (the json upload stuff). The release checks&creation should be its own block.

**Comment by chiichen** [comment]
> yeah sorry, i wanted to look at this for ages but there's always some bs coming in between 🙃
> 
> I hope i can give it a proper look on monday or tuesday (will be on vacation after that), but the thing i already noticed is that there's quite a lot of duplicate code now (the json upload stuff). The release checks&creation should be its own block.

I got u. I will do some changes later

**Comment by chiichen** [comment]
@FabianLars sorry for so many ci failures :) I believe it's working now.Do you have time to give it a proper look now?

**Comment by FabianLars** [comment]
no worries, thanks for resolving them :)

i'm working on another pr rn but will get to yours right after (today)

**Comment by FabianLars** [comment]
while trying to refactor it a bit, similar to how create-release has its own function i noticed that create-release basically contains the same code so we can simplify this quite a bit by just making releaseName optional there too. Gonna push a commit in a few :)

**Comment by FabianLars** [comment]
this way we also can use the existing draft lookup (not included in getReleaseByTag)

**Review by FabianLars — APPROVED** [review]

**Comment by ChristianPayne** [comment]
Hey all, just a heads up. My app was building fine up until one of the last two releases of this.
` - uses: tauri-apps/tauri-action@v0` was what I was set to and it was grabbing the most up to date major version 0.
I hard coded it back to two releases ago (` - uses: tauri-apps/tauri-action@v0.5.15`) and I can build again.

## What happened?
You can check out my releases [here](https://github.com/ChristianPayne/ennesults-rs/releases).
Essentially, my `v1.0.0` built fine with all files and a latest.json. For some unknown reason, one of the newer versions would not allow the .exe, .sig or latest.json files to be built.
With `v1.1.0`, I noticed that the files were not there and my updater was not working (no latest.json).
I hard coded the tauri-action script back to `v0.5.15` and my files are back. I am completely uneducated as to why or what was changed but I thought I would drop this note here because I see active work on this action.

Let me know if you guys need logs or more info or anything, I can help troubleshoot.

**Comment by chiichen** [comment]
> 大家好，提醒一下。我的应用程序构建得很好，直到最近两个版本发布。 ` - uses: tauri-apps/tauri-action@v0`这是我设定的，它抓取的是最新的主要版本 0。 我将其硬编码回两个版本之前（` - uses: tauri-apps/tauri-action@v0.5.15`），然后我就可以再次构建了。
> 
> ## 发生了什么？
> [您可以在此处](https://github.com/ChristianPayne/ennesults-rs/releases)查看我的版本。 基本上，我的版本`v1.0.0`构建良好，包含所有文件和 latest.json。由于某些未知原因，较新版本之一不允许构建 .exe、.sig 或 latest.json 文件。 使用`v1.1.0`，我注意到文件不存在，并且我的更新程序无法正常工作（没有 latest.json）。 我将 tauri-action 脚本硬编码回`v0.5.15`，我的文件又回来了。我完全不知道为什么或更改了什么，但我想在这里写下这条注释，因为我看到有人在积极开展这项行动。
> 
> 如果你们需要日志或更多信息或任何东西，请告诉我，我可以帮助排除故障。

Can you list out one of failed action(no latest.json) and one of success action?

Maybe you can turn to latest issues. Tons of feedbacks there (:
